src/main/java/seedu/bookmark/logic/parser/CliSyntax.java

```diff
@@ -11,10 +11,37 @@
     public static final Prefix PREFIX_TAG = new Prefix("t/");
     public static final Prefix PREFIX_TOTAL_PAGES = new Prefix("tp/");
     public static final Prefix PREFIX_BOOKMARK = new Prefix("b/");
+    public static final Prefix PREFIX_READING_PROGRESS = new Prefix("rp/");
     public static final Prefix PREFIX_PAGE = new Prefix("p/");
     public static final Prefix PREFIX_DEADLINE = new Prefix("d/");
     public static final Prefix PREFIX_NOTE_TITLE = new Prefix("n/");
     public static final Prefix PREFIX_NOTE_TEXT = new Prefix("txt/");
     public static final Prefix PREFIX_COMPLETED = new Prefix("c/");
     public static final Prefix PREFIX_NOT_COMPLETED = new Prefix("nc/");
+    public static final Prefix PREFIX_DUMMY = new Prefix("dummy/");
+
+    /**
+     * Returns a prefix based on the input string.
+     * @return Prefix based on the input string.
+     */
+    public static Prefix sortingPrefixGenerator(String prefix) {
+        Prefix result;
+        switch (prefix) {
+        case "n/":
+            result = PREFIX_NAME;
+            break;
+        case "g/":
+            result = PREFIX_GENRE;
+            break;
+        case "b/":
+            result = PREFIX_BOOKMARK;
+            break;
+        case "rp/":
+            result = PREFIX_READING_PROGRESS;
+            break;
+        default:
+            return PREFIX_DUMMY;
+        }
```

Comment:
Maybe it will be better if we throw an exception instead of defaulting to a dummy sorting choice?

Reply:
Exception would be thrown in SortCommandParser if prefix was invalid. But I'll change this to return null in the default case to make it less confusing.